From acadc50a2b1fcdd598a9df9f7c1e062054221804 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Fri, 10 Jun 2011 16:48:26 +0200
Subject: [PATCH 3/8] No dtv on Android

Signed-off-by: Jean-Baptiste Kempf <jb@videolan.org>
---
 modules/access/Modules.am |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/modules/access/Modules.am b/modules/access/Modules.am
index 2db7a4c..080b1e7 100644
--- a/modules/access/Modules.am
+++ b/modules/access/Modules.am
@@ -158,6 +158,7 @@ libdtv_plugin_la_CFLAGS = $(AM_CFLAGS)
 libdtv_plugin_la_LIBADD = $(AM_LIBADD)
 libdtv_plugin_la_DEPENDENCIES =
 
+if !HAVE_ANDROID
 if HAVE_LINUX
 libdtv_plugin_la_SOURCES += dtv/linux.c
 if HAVE_DVBPSI
@@ -167,6 +168,7 @@ libdtv_plugin_la_LIBADD += $(DVBPSI_LIBS)
 endif
 libvlc_LTLIBRARIES += libdtv_plugin.la
 endif
+endif
 
 if HAVE_WIN32
 # Work-around automake bug:
-- 
1.7.4.1

