From a2db58df1b91f5a112b7c6180e934e74953905c1 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Sun, 31 Jul 2011 01:12:58 +0200
Subject: [PATCH 7/8] Thread: no nanoclock_sleep on Android

---
 src/posix/thread.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/posix/thread.c b/src/posix/thread.c
index a898e28..973209f 100644
--- a/src/posix/thread.c
+++ b/src/posix/thread.c
@@ -997,7 +997,7 @@ mtime_t mdate (void)
  */
 void mwait (mtime_t deadline)
 {
-#if (_POSIX_TIMERS > 0)
+#if (_POSIX_TIMERS > 0) && !defined(__ANDROID__)
     vlc_clock_setup ();
     /* If the deadline is already elapsed, or within the clock precision,
      * do not even bother the system timer. */
@@ -1025,7 +1025,7 @@ void msleep (mtime_t delay)
     struct timespec ts = mtime_to_ts (delay);
 
     vlc_clock_setup ();
-#if (_POSIX_TIMERS > 0)
+#if (_POSIX_TIMERS > 0) && !defined(__ANDROID__)
     while (clock_nanosleep (vlc_clock_id, 0, &ts, &ts) == EINTR);
 
 #else
-- 
1.7.4.1

