From c55df8cf7000758ba28c0000d4a402e4034d948d Mon Sep 17 00:00:00 2001
Message-Id: <c55df8cf7000758ba28c0000d4a402e4034d948d.1579874292.git.nicolas@videolabs.io>
From: Francois Cartegnie <fcvlcdev@free.fr>
Date: Wed, 22 Jan 2020 18:58:32 +0100
Subject: [PATCH] access: dvdnav: dvd seek

---
 modules/access/dvdnav.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/modules/access/dvdnav.c b/modules/access/dvdnav.c
index 36638d7680..9415372b13 100644
--- a/modules/access/dvdnav.c
+++ b/modules/access/dvdnav.c
@@ -59,6 +59,8 @@
 
 
 #include <dvdnav/dvdnav.h>
+/* Expose without patching headers */
+dvdnav_status_t dvdnav_jump_to_sector_by_time(dvdnav_t *, uint64_t, int32_t);
 
 #include "../demux/mpeg/pes.h"
 #include "../demux/mpeg/ps.h"
@@ -602,6 +604,17 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
             return VLC_EGENERIC;
         }
 
+        case DEMUX_SET_TIME:
+        {
+            mtime_t i_time = va_arg( args, mtime_t );
+            if( dvdnav_jump_to_sector_by_time( p_sys->dvdnav,
+                                               i_time * 90 / 1000,
+                                               SEEK_SET ) == DVDNAV_STATUS_OK )
+                return VLC_SUCCESS;
+            msg_Err( p_demux, "can't set time to %" PRId64, i_time );
+            return VLC_EGENERIC;
+        }
+
         /* Special for access_demux */
         case DEMUX_CAN_PAUSE:
         case DEMUX_CAN_SEEK:
-- 
2.17.1

